cache/auto_refresh.go

```diff
@@ -235,7 +235,8 @@ func (w *autoRefresh) enqueueBatches(ctx context.Context) error {
 
 	for _, batch := range batches {
 		b := batch
-		w.workqueue.Add(&b)
+		logger.Debugf(ctx, "Enqueuing batch with id: %v", b[0].GetID())
+		w.workqueue.Add(b[0].GetID())
```

Comment:
This assumes that batches only have one item... I don't think this is true in all usage of AutoRefresh cache... if it's, I would rather deprecate the whole batching mechanism...
This implementation won't work for any batches > 1 item.

Reply:
In flytepropeller, batches always only have one item.
https://github.com/flyteorg/flytestdlib/blob/1285ff30b85a0046c1ad32bfd971e6b28783276d/cache/auto_refresh.go#L133-L140
Will clean up the code.